### DISCUSSION TITLE
opening ports

### DISCUSSION THREAD

**Discussion by StackTrace18** [mido/mido#582] [Q&A]
Hi, I'm a newbie on mido and am having the following problem (after quick search of discussions) and would appreciate some help.
the issue is ports not sending or receiving messages. 
I am on a windows 11. I have a physical midi instrument connected via usb. 

```python
mido.get_output_names()
>['Microsoft GS Wavetable Synth 0', 'USB Uno MIDI Interface 1']

out=mido.open_output("USB Uno MIDI Interface 1")
assert not out.closed #ok
out.send(Message('note_on',note=64,velocity=127)
#instrument doesn't receive message, and
mido.get_input_names()
['USB Uno MIDI Interface 0']
in = mido.open_input("USB Uno MIDI Interface 0") 
in.callback=printmessage
#no input messages received in callback
```

**Comment by rdoursenaud** [comment]
Hi and sorry for the late reply.

Thank you for trying out mido!

Since your interface is recognized and accepting messages, please make sure that your interface connections to the instrument are not reversed first:
- USB interface out -> instrument in
- instrument out -> USB interface in

Next, maybe include your `printmessage` function in the snippet.

Otherwise, your code looks good and should produce the expected results.
